src/dev-app/dev-app/dev-app-theme.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Inject, Injectable} from '@angular/core';
+import {OverlayContainer} from '@angular/cdk/overlay';
+import {DOCUMENT} from '@angular/common';
+
+const isDarkThemeKey = 'isDarkTheme';
+
+@Injectable({providedIn: 'root'})
+export class DevAppTheme {
+  readonly darkThemeClass = 'demo-unicorn-dark-theme';
+  private _isDark = false;
+
+  constructor(private _overlayContainer: OverlayContainer,
+              @Inject(DOCUMENT) private _document: Document) {
+    let isDark: string | null;
+    try {
+      isDark = localStorage.getItem(isDarkThemeKey);
+      if (isDark != null) {
+        this._isDark = isDark === 'true';
+        this._isDark ? this.applyDarkTheme() : this.applyLightTheme();
+      }
+    } catch (error) {
+      console.error(`Failed to read ${isDarkThemeKey} from localStorage: `, error);
+    }
+  }
+
+  get isDark(): boolean {
+    return this._isDark;
+  }
+
+  set isDark(value: boolean) {
+    if (value != null) {
+      this._isDark = value;
```

Comment:
we could skip the update if the `value` is different to `this._isDark`. what do you think?